### PR TITLE
chore(mypy): enable `truthy-bool` error code

### PR DESCRIPTION
**Pull request by sobolevn**
Docs: https://mypy.readthedocs.io/en/stable/error_code_list2.html#check-that-expression-is-not-implicitly-true-in-boolean-context-truthy-bool

I am using `# type: ignore[truthy-bool]` for `.components` check, because I see that it is treated specially for some reason (like `pragma: no cover`). Is this correct?

I also removed configs that are part of the `--strict` flag from settings.

### CODE REVIEW

**Comment by sobolevn** [comment]
Please, take a look at these lines:
Снимок экрана 2024-10-17 в 11 43 46

Looks like `else` is never actually reached. Should I remove this branch?

File: pyproject.toml [hunk 1]
```diff
@@ -253,21 +254,19 @@ enable_error_code = [
...
 disallow_any_generics = false
-disallow_untyped_decorators = true
-implicit_reexport = false
 show_error_codes = true
 strict = true
-warn_redundant_casts = true
-warn_return_any = true
 warn_unreachable = true
-warn_unused_configs = true
-warn_unused_ignores = true
```
Comment (provinzkraut) [inline pyproject.toml:264]: Why have these been removed?

**Comment by provinzkraut** [comment]
> Looks like `else` is never actually reached. Should I remove this branch?

Seems to be a missing test case?

File: pyproject.toml (hunk 1 above)
Comment (Alc-Alc) [reply pyproject.toml:264]: from PR description
> I also removed configs that are part of the --strict flag from settings.

**Comment by provinzkraut** [comment]
>     * Extra `else` branch in this `if` ?

Isn't it the `elif` branch that's extra then, and should be an `else` instead?

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**